packages/mgt-chat/src/components/ChatList/ChatList.tsx

```diff
@@ -61,101 +60,102 @@ const useStyles = makeStyles({
-        if (state.status === 'chat threads loaded' && props.onLoaded) {
-          const markAllAsRead = {
-            displayText: 'Mark all as read',
-            onClick: () => markAllThreadsAsRead(state.chatThreads)
-          };
-          // clone the menuItems array
-          const updatedMenuItems = [...menuItems];
-          updatedMenuItems.unshift(markAllAsRead);
-          setMenuItems(updatedMenuItems);
-          props.onLoaded();
-        }
+    // handle state changes
+    chatListClient.onStateChange(setChatListState);
+    chatListClient.onStateChange(state => {
+      if (state.status === 'chat threads loaded' && onLoaded) {
+        onLoaded();
+      }
 
-        if (state.status === 'server connection established') {
-          setHeaderBannerMessage(''); // reset
-        }
+      if (state.status === 'server connection established') {
+        setHeaderBannerMessage(''); // reset
+      }
 
-        if (state.status === 'creating server connections') {
-          setHeaderBannerMessage('Connecting...');
-        }
+      if (state.status === 'creating server connections') {
+        setHeaderBannerMessage('Connecting...');
+      }
 
-        if (state.status === 'server connection lost') {
-          // this happens when we lost connection to the server and we will try to reconnect
-          setHeaderBannerMessage('We ran into a problem. Reconnecting...');
-        }
-      });
-
-      chatListClient.onChatListEvent(handleChatListEvent);
-      return () => {
-        // log state of chatlistclient for debugging purposes
-        log(chatListClient.getState());
-        chatListClient.offStateChange(setChatListState);
-        chatListClient.offChatListEvent(handleChatListEvent);
-        chatListClient.tearDown();
+      if (state.status === 'server connection lost') {
+        // this happens when we lost connection to the server and we will try to reconnect
         setHeaderBannerMessage('We ran into a problem. Reconnecting...');
-      };
-    }
-  }, [chatListClient]);
+      }
+    });
 
-  const markAllThreadsAsRead = (chatThreads: GraphChat[]) => {
-    const readChatThreads = chatThreads.map(c => c.id!);
-    const markedChatThreads = chatListClient?.markChatThreadsAsRead(readChatThreads);
-    if (markedChatThreads) {
-      chatListClient?.cacheLastReadTime(markedChatThreads);
-      props.onAllMessagesRead(markedChatThreads);
-    }
-  };
+    // handle chat list events
+    chatListClient.onChatListEvent(handleChatListEvent);
+
+    // tear down
+    return () => {
+      // log state of chatlistclient for debugging purposes
+      log(chatListClient.getState());
+      chatListClient.offStateChange(setChatListState);
+      chatListClient.offChatListEvent(handleChatListEvent);
+      chatListClient.tearDown();
+      setHeaderBannerMessage('We ran into a problem. Please close or refresh.');
+    };
+  }, [chatListClient, onMessageReceived, onAllMessagesRead, onLoaded]);
```

Comment:
This onAllMessagesRead dependency can be removed.
```suggestion
  }, [chatListClient, onMessageReceived, onLoaded]);
```

Reply:
good catch